tests/test_common.py

```diff
@@ -0,0 +1,298 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+"""
+Common utils for testing.
+"""
+
+import base64
+import os
+import time
+from subprocess import PIPE, Popen
+from typing import Any, Dict, List
+
+import httpx
+import pytest
+import typing_extensions
+from loguru import logger
+
+
+class Sandbox:
+    """
+    Run the sandbox with lskv loaded.
+    """
+
+    def __init__(self, http2: bool):
+        self.nodes = 1
+        self.port = 8000
+        self.http_version = 2 if http2 else 1
+        self.proc = None
+
+    def __enter__(self):
+        self.proc = self.spawn()
+
+    def __exit__(
+        self, ex_type, ex_value, ex_traceback
+    ) -> typing_extensions.Literal[False]:
+        if self.proc:
+            logger.info("terminating store process")
+            self.proc.terminate()
+            # give it a second to shutdown
+            time.sleep(1)
+            if not self.proc.poll():
+                # process is still running, kill it
+                logger.info("killing store process")
+                self.proc.kill()
+            self.proc.wait()
+            logger.info("stopped")
+
+        return False
+
+    def wait_for_ready(self) -> bool:
+        """
+        Wait for the datastore to be ready to accept requests.
+        """
+        return self._wait_for_ready(self.port)
+
+    def _wait_for_ready(self, port: int, tries=30) -> bool:
+        client = self.etcdctl_client()
+        client += ["get", "missing key"]
+        if self.http_version == 1:
```

Comment:
Why is it skipped if `http_version == 2`? Also, you may want to use `httpx` here rather than `curl` for consistency, unless this provides better debugging?

Reply:
This was just a copy of what is used in the benchmarking. For http2 we can use the actual etcdctl client. I'll probably keep it using curl for now for consistency with running the `etcdctl` for http2